### PR TITLE
Updated the query to return all relevant transactions

### LINKED ISSUE #9606: Transactions API Should Return All Relevant Transactions
### Problem

`parent_consensus_timestamp` is only set on following children. #4672 made changes to exclude synthetic transactions and inadvertently excludes transactions that precede another transaction. For instance, a `CRYPTOTRANSFER` that creates a  `CRYPTOCREATEACCOUNT` transaction. These `CRYPTOCREATEACCOUNT` transactions will have a non-zero `nonce` and will not have a `parent_consensus_timestamp` so the query adjustment in the mentioned pr will exclude these transactions in the response.

### Solution

- [ ] Determine other transaction types that may be effected
- [ ] Adjust query so that all relevant transactions are returned

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by HarshSawarkar**
**Description**:

Modified the query to return all relevant transaction
**Related issue(s)**:

Fixes #9606 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jnels124 — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/transactions.js [hunk 1]
```diff
@@ -343,8 +343,12 @@ const getTransferDistinctTimestampsQuery = (
 };
 
 // the condition to exclude synthetic transactions attached to a user submitted transaction
+const transactionTypesToInclude = ['CRYPTOAPPROVEALLOWANCE', 'CRYPTOTRANSFER', 'CRYPTODELETEALLOWANCE'];
```
[the remark this answers is not in the document]
Comment (HarshSawarkar) [reply hedera-mirror-rest/transactions.js]: Updated.

File: hedera-mirror-rest/transactions.js [hunk 2]
```diff
@@ -344,7 +345,8 @@ const getTransferDistinctTimestampsQuery = (
 
 // the condition to exclude synthetic transactions attached to a user submitted transaction
 const transactionByPayerExcludeSyntheticCondition = `${Transaction.getFullName(Transaction.NONCE)} = 0 or
-  ${Transaction.getFullName(Transaction.PARENT_CONSENSUS_TIMESTAMP)} is not null`;
+  ${Transaction.getFullName(Transaction.PARENT_CONSENSUS_TIMESTAMP)} is not null 
+  or ${Transaction.getFullName(Transaction.TYPE)} in (${typesToInclude.map((type) => `'${type}'`).join(', ')})`;
```
Comment (jnels124) [inline hedera-mirror-rest/transactions.js:349]: Once other comments are addressed, `.map` can be removed. Also, we should only add condition if array is not empty.

File: hedera-mirror-rest/config/application.yml [hunk 3]
```diff
@@ -41,6 +41,10 @@ hedera:
           username: metrics
           uriPath: "/swagger"
         ipMetrics: false
+      transactions:
```
Comment (jnels124) [inline hedera-mirror-rest/config/application.yml:44]: This should be the protoId of these types. See `TransactionType.java`.

Comment (jnels124) [reply hedera-mirror-rest/config/application.yml:44]: Also, this property should be added to `configuration.md`

File: hedera-mirror-rest/transactions.js (hunk 2 above)
Comment (jnels124) [reply hedera-mirror-rest/transactions.js:349]: Should update the `hedera-mirror-rest/__tests__/specs/transactions/account-id.json` spec test to include these types of transactions.

**Review by jnels124 — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/config/application.yml (hunk 3 above)
Comment (HarshSawarkar) [reply hedera-mirror-rest/config/application.yml]: Updated

File: hedera-mirror-rest/transactions.js (hunk 2 above)
Comment (HarshSawarkar) [reply hedera-mirror-rest/transactions.js]: Updated

File: hedera-mirror-rest/config.js [hunk 4]
```diff
@@ -188,6 +188,13 @@ const parseNetworkConfig = () => {
   }
 };
 
+const parseTransactionConfig = () => {
```
Comment (jnels124) [inline hedera-mirror-rest/config.js:191]: can parse this in `parseQueryConfg`

**Pull request closed without merging**